evcouplings/utils/pipeline.py

```diff
@@ -472,6 +476,89 @@ def _handler(signal_, frame):
         raise
 
 
+def calculate_memory_requirements(config):
+    """
+
+    Calculates the memory requirements in MB based on the pipeline configuration
+
+    memory_in_MB = (1/2 * q^2 * (L - 1) * L  + q * L) / 12500
+
+    with:
+        - q: states of Potts model
+        - L: Sequence length
+
+
+    Parameters
+    ----------
+    config : dict
+        A dictionary holding the pipeline configurations
+
+    Returns
+    -------
+    int
+        RAM requirements in MB
+
+    Raises
+    ------
+    ResourceError
+    """
+
+    # gather alignment stages (monomer has only one, complex has 2)
+    stages = filter(lambda x: "align" in x, config["stages"])
+    L = 0
+
+    # extract state information
+    # if alphabet is not explicitly defined assume amino acid alphabet + gap
+    q = len(config["couplings"]["alphabet"]) if config["couplings"]["alphabet"] else 21
```

Comment:
* Alphabet can be defined using a shortcut like "aa" or "dna" as well, please refer to respective code in couplings protocol
* If using default, please use respective constant defining aa alphabet also defined in couplings protocol